Makefile.toml

```
@@ -407,6 +429,77 @@ ln -sfn "${PUBLISH_REPO_OUTPUT_DIR##*/}" "${PUBLISH_REPO_OUTPUT_DIR%/*}/latest"
 '''
 ]
 
+[tasks.validate-repo]
+dependencies = ["publish-setup-without-key", "publish-tools"]
+script_runner = "bash"
+script = [
+'''
+set -e
+
+export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
+
+pubsys \
+   --infra-config-path "${PUBLISH_INFRA_CONFIG_PATH}" \
+   \
+   validate-repo \
+   \
+   --repo "${PUBLISH_REPO}" \
+   --arch "${BUILDSYS_ARCH}" \
+   --variant "${BUILDSYS_VARIANT}" \
+   \
+   --root-role-path "${PUBLISH_REPO_ROOT_JSON}" \
+   --percent-target-files "${VALIDATE_PERCENT_TARGETS}"
+'''
+]
+
+[tasks.check-repo-expirations]
+dependencies = ["publish-setup-without-key", "publish-tools"]
+script_runner = "bash"
+script = [
+'''
+set -e
+
+export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
+
+pubsys \
+   --infra-config-path "${PUBLISH_INFRA_CONFIG_PATH}" \
+   \
+   check-repo-expirations \
+   \
+   --repo "${PUBLISH_REPO}" \
+   --arch "${BUILDSYS_ARCH}" \
+   --variant "${BUILDSYS_VARIANT}" \
+   \
+   --root-role-path "${PUBLISH_REPO_ROOT_JSON}" \
+   --expirations-between-now-and "in ${EXPIRING_WITHIN}"
```

```suggestion
   --expiring-within "${EXPIRING_WITHIN}"
```
(`parse_datetime` doesn't require the "in")

(I think "expirations between now and" is OK within the code, because the `parse_datetime` call effectively transforms the meaning of the variable.  "expiration limit" is also good, which you use later in the code.  It'd be good to standardize on one within the Rust code, though.)